site/pages/[locale]/token-approvals.js

```diff
@@ -14,6 +15,25 @@ import { useTokenInput } from '../../hooks/useTokenInput'
 import { fromUnit, toUnit } from '../../utils'
 import CallToAction from '../../components/CallToAction'
 
+const decimalRegex = /^(([1-9][0-9]*)?[0-9](\.[0-9]*)?|\.[0-9]+)$/
+const infiniteSymbol = '∞'
+
+const isInfinite = function (allowance, token) {
+  if (!allowance || !token) {
+    return false
+  }
+
+  if (allowance === infiniteSymbol) {
+    return true
+  }
```

Comment:
The logic should not operate on visual/UI state. The infinity symbol shall only be a UI representation of the state of the form. Let's revisit this.

Reply:
fixed, using component state `isInfinity` now